cincinnati/src/lib.rs

```diff
@@ -237,14 +238,77 @@ impl Serialize for Graph {
     }
 }
 
+impl PartialEq for Graph {
+    fn eq(&self, other: &Graph) -> bool {
+        use daggy::petgraph::visit::IntoNeighbors;
+
+        let asc_order_release_by_version = {
+            use std::cmp::Ordering::{self, *};
+
+            |a: &&Release, b: &&Release| -> Ordering {
+                if a.version() < b.version() {
+                    Less
+                } else if a.version() == b.version() {
+                    Equal
+                } else {
+                    Greater
+                }
+            }
+        };
+
+        // Look through all nodes in self
+        self.dag.node_references().all(|node_ref| {
+            let dag_other = &other.dag;
+            let node_index = node_ref.0;
+            let release = node_ref.1;
+
+            // For each node in self, look through all nodes in other and find a
+            // match
+            dag_other
+                .node_references()
+                .filter(|node_ref_other| {
+                    let node_index_other = node_ref_other.0;
+                    let release_other = node_ref_other.1;
+
+                    // Node matches
+                    release == release_other && {
+                        let (mut neighbor_releases, mut neighbors_releases_other) = (vec![], vec![]);
+                        // Determine neighbors of own and matching node
+                        self.dag
+                            .neighbors(node_index)
+                            .zip(dag_other.neighbors(node_index_other))
```

Comment:
I'm not sure about my understanding, so please double-check the following assertion. When `zip` combines the two iterators, the resulting length is the shortest one. Thus if there a matching on a strict subset, this may still think the two iterators are equals.

Reply:
Nice catch, your suspicion was correct on this one. The zipper ends as soon as one of the iterators returns `None`. I'll fix this up

Reply:
I went with the trivial solution and prepended the iteration with a count assertion